addons/mail/models/mail_thread.py

```diff
@@ -374,6 +375,39 @@ def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu
     # Technical methods / wrappers / tools
     # ------------------------------------------------------
 
+    def _replace_local_links(self, html, base_url=None):
+        """ Replace local links by absolute links. It is required in various
+        cases, for example when sending emails on chatter or sending mass
+        mailings. It replaces
+
+         * href of links (mailto will not match the regex)
+         * src of images (base64 hardcoded data will not match the regex)
+         * styling using url like background-image: url
+
+        It is done using regex because it is shorten than using an html parser
+        to create a potentially complex soupe and hope to have a result that
+        has not been harmed.
+        """
+        if not html:
+            return html
+
+        from odoo.tools import ustr
+        html = ustr(html)
+
+        def _sub_relative2absolute(match):
+            # compute here to do it only if really necessary + cache will ensure it is done only once
+            # if not base_url
+            if not _sub_relative2absolute.base_url:
+                _sub_relative2absolute.base_url = self.env["ir.config_parameter"].sudo().get_param("web.base.url")
+            return match.group(1) + urls.url_join(_sub_relative2absolute.base_url, match.group(2))
+
+        _sub_relative2absolute.base_url = base_url
+        html = re.sub(r'(<img(?=\s)[^>]*\ssrc=")(/[^/][^"]+)', _sub_relative2absolute, html)
+        html = re.sub(r'(<a(?=\s)[^>]*\shref=")(/[^/][^"]+)', _sub_relative2absolute, html)
+        html = re.sub(r'(<[^>]+\bstyle="[^"]+\burl\(\'?)(/[^/\'][^\'")]+)', _sub_relative2absolute, html)
```

Comment:
Using triple-quoted strings would avoid necessity of escaping quotes, and with the VERBOSE flag would also allow line-breaks and comments inside the regex literal to explain each regex & the more complicated bits

Reply:
I don't care I copy-and-paster them :D .

Reply:
I added triple quotes. But I did not add comments as I hate regexes.